clients/src/test/java/org/apache/kafka/common/message/MessageTest.java

```diff
@@ -465,10 +465,18 @@ public void testOffsetFetchVersions() throws Exception {
         OffsetFetchRequestData allPartitionData = new OffsetFetchRequestData()
                                                       .setGroupId(groupId)
                                                       .setTopics(null);
+
+        OffsetFetchRequestData requireStableData = new OffsetFetchRequestData()
+                                                       .setGroupId(groupId)
+                                                       .setTopics(null)
+                                                       .setRequireStable(true);
+
         for (short version = 0; version <= ApiKeys.OFFSET_FETCH.latestVersion(); version++) {
+            final short finalVersion = version;
             if (version < 2) {
-                final short finalVersion = version;
                 assertThrows(SchemaException.class, () -> testAllMessageRoundTripsFromVersion(finalVersion, allPartitionData));
+            } else if (version < 7) {
+                assertThrows(UnsupportedVersionException.class, () -> testAllMessageRoundTripsFromVersion(finalVersion, requireStableData));
             } else {
                 testAllMessageRoundTripsFromVersion(version, allPartitionData);
             }
```

Comment:
Not sure if I understand this test -- isn't `finalVersion` always the same as `version`? Also, it seems that we don't cover all combination? If version is eg, 5, we only test that `requireStableData` throws, but we don't test that `allPartitionData` does not throw.

Reply:
`finalVersion` is needed as lambda function requires final val to pass in, and second point is correct, will breakdown this else-if logic into if-if

Reply:
Ah... thanks.